src/c++/perf_analyzer/client_backend/mock_client_backend.h

```diff
@@ -118,6 +118,15 @@ class MockInferResult : public InferResult {
     return Error::Success;
   }
 
+  Error IsFinalResponse(bool* is_final_response) const override
```

Comment:
Why is there an implementation in the mock class?

Reply:
`InferContext::AsyncCallbackFuncImpl()` calls `InferResult::IsFinalResponse()`, so the mock class needs to define its own implementation so that no error is returned and the value gets set to true.

I rewrote some of the relevant `InferContext::AsyncCallbackFuncImpl()` code to be more clear.